screens/map/MapScreen.js

```diff
@@ -35,198 +36,158 @@ import {
 
 const snapPoints = [185, 325, 488];
 
-const deltas = {
-  latitudeDelta: 0.01,
-  longitudeDelta: 0.01,
-};
-
-export default class MapScreen extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      locationErrorMsg: null,
-      region: initialRegion,
-      stores: null,
-      store: null,
-      storeProducts: null,
-      showDefaultStore: false,
-    };
-  }
-
-  async componentDidMount() {
-    // We get current location first, since we need to use the lat/lon found in _populateIntitialStoresProducts
-    await this._findCurrentLocation();
-    await this._populateInitialStoresProducts();
-  }
+export default function MapScreen(props) {
+  const [locationPermissions, setLocationPermissions] = useState(null);
+  const [region, setRegion] = useState(initialRegion);
+  const [stores, setStores] = useState([]);
+  const [initialLocation, setInitialLocation] = useState(null);
+  const [currentStore, setCurrentStore] = useState(null);
+  const [storeProducts, setStoreProducts] = useState([]);
+  const [showDefaultStore, setShowDefaultStore] = useState(true);
```

Comment:
Your goal should be to be able to knock all of these hooks out by composing them with your custom hook 😄 Give that a shot and see how it feels.

Reply:
When you say 'all' do you really mean *all*?  I knocked out most of them, but I'm still left with `useState` for `currentStore` and `region` (which is the user's location panning/zooming around on the map). It still makes sense to keep these right?

Reply:
Yes, you're right — makes sense to keep `currentStore` and `region` since they're component-specific!